wagtail/admin/tests/test_menu.py

```diff
@@ -368,3 +368,65 @@ def test_menu_items_have_names(self):
         # We want the name to be consistent across languages, so this test will
         # fail if the label is translated.
         self.assertFalse(expected - names)
+
+    def test_submenu_items_have_names(self):
+        # Delete the registered_menu_items cache
+        try:
+            del admin_menu.registered_menu_items
+        # The cache may not be created yet if the test is run in isolation
+        except AttributeError:
+            pass
+
+        # Getting only the submenu items
+        submenu_items_to_check = set()
+        for item in admin_menu.registered_menu_items:
+            if not hasattr(item, "menu"):
+                continue
+
+            for subitem in item.menu.registered_menu_items:
+                submenu_items_to_check.add(subitem)
+
+        # Generate the submenu items using a different language
+        with translation.override("fr"):
+            names = {item.name for item in submenu_items_to_check}
```

Comment:
I thought this as well and the tests still went through. Then I noticed something a bit odd:

I used to run the tests with the following command: `python runtests.py wagtail.admin.tests.test_menu` and even if I comment out my change for the users hook, the test still passes (the name I get as an output is still `users`). Then I tried to run this test case only: `python runtests.py wagtail.admin.tests.test_menu.test_submenu_items_have_names`, and the test fails as we would expect (it outputs `utilisateurs`). Now I have a doubt about `gettext_lazy`: could it be possible that the very first evaluation of the lazy object has some sort of internal caching? 

This could be an explanation to a potential cross testcase pollution, which I suspect. On the other hand I tried to replicate the same behaviour on the first level item names test and it seems to behave as expected. I'll dig a bit more into this

Reply:
`registered_menu_items` is a cached property, so if another test has already evaluated the value with a different active locale, it can produce a different result. One way to make the test deterministic is to clear the cache (with a `del` in a `try`/`except` block) before accessing the value, just like what has been done for `admin_menu.registered_menu_items` in the lines above this block.